dask/array/rechunk.py

```diff
@@ -45,7 +45,15 @@ def _breakpoints(cumold, cumnew):
     >>> _breakpoints(new[1], old[1])
     (('n', 0), ('o', 0), ('n', 2), ('n', 4), ('n', 5), ('o', 5))
     """
-    return tuple(sorted(cumold + cumnew, key=itemgetter(1)))
+    merged = cumold + cumnew
+    if any(v != v for _, v in merged):
```

Comment:
Rather than complicate all subroutines to support unknown dims, perhaps it's easier to modify the callers?